src/app/app-services/authorization.service.ts

```diff
@@ -1,45 +1,54 @@
 import { Injectable, Inject } from '@angular/core';
 import { User } from '@app/app-models';
+import { HttpClient } from '@angular/common/http';
+import { switchMap, tap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthorizationService {
 
-  private readonly KEY = 'CURRENT_USER';
+  readonly USER = 'CURRENT_USER';
+  readonly TOKEN = 'TOKEN';
+  token?: string;
 
-  constructor(@Inject('Storage') private readonly localStorage: Storage) {
+  constructor(@Inject('Storage') private readonly localStorage: Storage,
+    @Inject('BASE_URL') private readonly BASE_URL: string,
+    private readonly httpClient: HttpClient) {
```

Comment:
One more tip to format methods with many arguments:

```typescript
constructor(
    @Inject('Storage') private readonly localStorage: Storage,
    @Inject('BASE_URL') private readonly BASE_URL: string,
    private readonly httpClient: HttpClient
) {}
```

Reply:
Ditto